src/AspNetCore/Src/aspnetcoreconfig.cxx

```diff
@@ -8,11 +8,16 @@ ASPNETCORE_CONFIG::~ASPNETCORE_CONFIG()
     //
     // the destructor will be called once IIS decides to recycle the module context (i.e., application)
     //
+    // shutting down core application first
+    if (ASPNETCORE_APPLICATION::GetInstance() != NULL) {
```

Comment:
this will cause problem in case of more than one applications

Reply:
Do you mean mul

Reply:
so our hosting mode is one application per apppool? it's hard to force this rule and outproc does support multiple applications inside same apppool

Reply:
> Have we try it without appdomain and know where the blocker is?
I mean either way the app will fail to load as the static ASPNETCORE_APPLICATION is already set.

Reply:
@pan-wang There are no appdomains on .NET Core.